x-pack/plugins/cases/public/components/create/templates.tsx

```diff
@@ -23,29 +23,39 @@ import { TEMPLATE_HELP_TEXT, TEMPLATE_LABEL } from './translations';
 interface Props {
   isLoading: boolean;
   templates: CasesConfigurationUI['templates'];
-  onTemplateChange: (caseFields: CasesConfigurationUITemplate['caseFields']) => void;
+  templateToSelect?: CasesConfigurationUI['templates'][number];
+  onTemplateChange: ({
+    caseFields,
+    key,
+  }: Pick<CasesConfigurationUITemplate, 'caseFields' | 'key'>) => void;
 }
 
 export const TemplateSelectorComponent: React.FC<Props> = ({
   isLoading,
   templates,
+  templateToSelect,
   onTemplateChange,
 }) => {
   const [selectedTemplate, onSelectTemplate] = useState<string>();
   const isSmallScreen = useIsWithinMaxBreakpoint('s');
 
+  if (templateToSelect && !selectedTemplate) {
```

Comment:
it's feels weird to me having to call some update functions depending on the props, shouldn't the parent components be handling this already? I think the issue here is having the selectedTemplate state here, it sounds like it should be stored in the parent component and this one just be responsible of the UI

Reply:
I agree it feels weird but I don't know if the `selectedTemplate` here is the problem. This component has the `EuiSelect` so we need some of this logic here.

What if we pass `templateToSelect` from props as the default on line 39 and do the initial call to `onTemplateChange` on the parent component? If that works we could at least remove this block

Reply:
This change was already in my stash and I forgot to apply 🤦‍♀️  [here](https://github.com/elastic/kibana/pull/190701/commits/c55a653aa32136db27b898aa532cfa3f8f37eb9b#diff-446bd7f9149b5e7a474d540815af188406428ef09d10e12278589fd486693196)